source/mongodb-handshake/handshake.rst

```diff
@@ -224,20 +224,75 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
-Limitations
-===========
+Supporting Wrapping Libraries
+=============================
 
-The entire metadata BSON document MUST NOT exceed 512 bytes. This includes all
-BSON overhead.  The ``client.application.name`` cannot exceed 128 bytes.  MongoDB
-will return an error if these limits are not adhered to, which will result in
-handshake failure. Drivers MUST validate these values and truncate driver
-provided values if necessary.
+Drivers MUST allow libraries which wrap the driver to append to the client
+metadata generated by the driver. The following class definition defines the
+options which MUST be supported:
+
+.. code:: typescript
+
+    class DriverInfoOptions {
+        /**
+        * The name of the library wrapping the driver.
+        */
+        name: String;
+
+        /**
+        * The version of the library wrapping the driver.
+        */
+        version: Optional<String>;
+
+        /**
+        * Optional platform information for the wrapping driver.
+        */
+        platform: Optional<String>;
+    }
+
+
+Note that how these options are provided to a driver is left up to the implementor.
+
+If provided, these options MUST NOT replace the values used for metadata generation.
+The provided options MUST be appended to their respective fields, and be delimited by
+a ``|`` character. For example, when `Motor <https://docs.mongodb.com/ecosystem/drivers/motor/>`_
```

Comment:
Might have been missed from a previous comment I made, but should we prohibit the three options from containing a delimiter character (noting below that you don't require a `|` and allow drivers to keep whatever they may already be using)? Alternatively, we can ignore this edge case.

I'm OK with either and just wanted to ensure that this question was considered.

Reply:
added a note requiring validation of this

Reply:
/cc @ShaneHarvey @jyemin, you both already approved but I added another `MUST NOT` here and `MUST throw an error`, which may be contentious

Reply:
Sounds good to me.

Reply:
Me too.

In the future don't force push.  Makes it hard to see diffs after initial review.  You can squash before merging.  :)